### PR TITLE
Run shellcheck on all scripts

### PR DESCRIPTION
**Pull request by EmilienM**
This is important to maintain good quality of our scripts, so we need to
test them all.

This patch enables shellcheck for all *.sh files and also address the
suggestions from shellcheck to make it happy.

### CODE REVIEW

**Comment by EmilienM** [comment]
Also look https://github.com/openshift/release/pull/21468

File: Makefile [hunk 1]
```diff
@@ -20,14 +20,6 @@ server:
...
-	shellcheck \
-		./clean-ci-resources.sh \
-		./list-clusters \
-		./metrics.sh \
-		./pull-artifacts \
-		./refresh_rhcos.sh \
-		./report.sh \
-		./server.sh \
-		./stale
+	find . -type f -name "*.sh" | xargs --verbose --no-run-if-empty shellcheck
```
Comment (pierreprinetti) [inline Makefile:23]: You are missing `list-clusters`, `pull-artifacts` and  `stale`. Shall we rename them?

**Comment by mandre** [comment]
If you get around updating your change, can you also fix the typo in your commit title, spellcheck -> shellcheck?

File: Makefile (hunk 1 above)
Comment (EmilienM) [reply Makefile:23]: yes, I'll take care of it

**Review by mandre — COMMENTED** [review]
/lgtm

**Pull request merged**